consensus/ibft/src/integration-test/java/tech/pegasys/pantheon/consensus/ibft/support/TestContextFactory.java

```diff
@@ -0,0 +1,264 @@
+import tech.pegasys.pantheon.consensus.ibft.RoundTimer;
+import tech.pegasys.pantheon.consensus.ibft.blockcreation.IbftBlockCreatorFactory;
+import tech.pegasys.pantheon.consensus.ibft.blockcreation.ProposerSelector;
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.MessageFactory;
+import tech.pegasys.pantheon.consensus.ibft.statemachine.IbftBlockHeightManagerFactory;
+import tech.pegasys.pantheon.consensus.ibft.statemachine.IbftController;
+import tech.pegasys.pantheon.consensus.ibft.statemachine.IbftFinalState;
+import tech.pegasys.pantheon.consensus.ibft.statemachine.IbftRoundFactory;
+import tech.pegasys.pantheon.consensus.ibft.validation.MessageValidatorFactory;
+import tech.pegasys.pantheon.crypto.SECP256K1.KeyPair;
+import tech.pegasys.pantheon.ethereum.ProtocolContext;
+import tech.pegasys.pantheon.ethereum.chain.MutableBlockchain;
+import tech.pegasys.pantheon.ethereum.core.Address;
+import tech.pegasys.pantheon.ethereum.core.AddressHelpers;
+import tech.pegasys.pantheon.ethereum.core.Block;
+import tech.pegasys.pantheon.ethereum.core.BlockBody;
+import tech.pegasys.pantheon.ethereum.core.BlockHeader;
+import tech.pegasys.pantheon.ethereum.core.BlockHeaderTestFixture;
+import tech.pegasys.pantheon.ethereum.core.Hash;
+import tech.pegasys.pantheon.ethereum.core.MiningParameters;
+import tech.pegasys.pantheon.ethereum.core.PendingTransactions;
+import tech.pegasys.pantheon.ethereum.core.Util;
+import tech.pegasys.pantheon.ethereum.core.Wei;
+import tech.pegasys.pantheon.ethereum.db.WorldStateArchive;
+import tech.pegasys.pantheon.ethereum.mainnet.BlockHeaderValidator;
+import tech.pegasys.pantheon.ethereum.mainnet.ProtocolSchedule;
+import tech.pegasys.pantheon.util.bytes.BytesValue;
+import tech.pegasys.pantheon.util.uint.UInt256;
+
+import java.time.Clock;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.Iterables;
+
+public class TestContextFactory {
+
+  private static class ControllerAndState {
+
+    private IbftController controller;
+    private IbftFinalState finalState;
+
+    public ControllerAndState(final IbftController controller, final IbftFinalState finalState) {
+      this.controller = controller;
+      this.finalState = finalState;
+    }
+
+    public IbftController getController() {
+      return controller;
+    }
+
+    public IbftFinalState getFinalState() {
+      return finalState;
+    }
+  }
+
+  public static final int EPOCH_LENGTH = 10_000;
+  public static final int BLOCK_TIMER_SEC = 3;
+  public static final int ROUND_TIMER_SEC = 12;
+
+  public static TestContext createTestEnvWithUtcClock(
+      final int validatorCount, final int indexOfFirstLocallyProposedBlock) {
+    return createTestEnvironment(
+        validatorCount, indexOfFirstLocallyProposedBlock, Clock.systemUTC());
```

Comment:
This is very much a knee-jerk reaction, but do we really have to support using a real clock in a test? It just seems to be asking for intermittency...

Reply:
done.